core/client/fs/src/main/java/alluxio/client/block/stream/GrpcDataWriter.java

```diff
@@ -154,6 +155,10 @@ private GrpcDataWriter(FileSystemContext context, final WorkerNetAddress address
           .setFallback(alreadyFallback).build();
       builder.setCreateUfsBlockOptions(ufsBlockOptions);
     }
+    // check if we need to pin block on create
+    if (options.getWriteType() == WriteType.ASYNC_THROUGH) {
+      builder.setPinned(true);
+    }
```

Comment:
Should we use the `builder.setPinned(options.getWriteType() == ...)` pattern here as well?